### PR TITLE
Support JWKs with no algorithm

### PR DESCRIPTION
**Pull request by tanner0101**
Adds support for JSON Web Keys that do not specify an algorithm (#34).

JWKs are now stored directly in `JWTSigners` via the `use(jwk:)` method. When a request to verify a JWT is received that matches a JWK, both the JWT and the JWK's algorithm header is checked. A signer is then created on the fly to handle the request.

### CODE REVIEW

**Comment by tanner0101** [comment]
Related to https://github.com/vapor/jwt-kit/pull/22

File: Sources/JWTKit/Signing/JWTSigners.swift [hunk 1]
```diff
@@ -21,24 +23,59 @@ public final class JWTSigners {
...
+        _ signer: JWKSigner,
+        kid: JWKIdentifier
+    ) {
+        self.jwkStorage[kid] = signer
+    }
+
     /// Gets a signer for the supplied `kid`, if one exists.
-    public func get(kid: JWKIdentifier? = nil) -> JWTSigner? {
+    public func get(kid: JWKIdentifier? = nil, alg: String? = nil) -> JWTSigner? {
+        print(kid, alg)
```
Comment (jdmcd) [inline Sources/JWTKit/Signing/JWTSigners.swift:63]: Print line here

**Comment by tanner0101** [comment]
Note: Ideally this fix would be achieved by a more involved refactor that would reduce code duplication. This is not possible for this release, so it will be targeted at the next one: https://github.com/vapor/jwt-kit/issues/35

**Comment by tanner0101** [comment]
These changes are now available in [4.0.0-rc.2](https://github.com/vapor/jwt-kit/releases/tag/4.0.0-rc.2)

**Pull request merged**